### DISCUSSION TITLE
pipenv fails to generate lock file for flask 2.0.1 and 2.1.1 even when using tilde operator

### DISCUSSION THREAD

**Discussion by vishal180618** [pypa/pipenv#5425] [Show and tell]
Problem: even though I've used `~` in dependency and both flask versions are changed in minor versions so it should be able to generate `Pipfile.lock ` using `pipenv lock`.

difference bw the pipfile.lock generated individually for both flask versions: https://www.diffchecker.com/GngEExsR  

Error:
```
CRITICAL:pipenv.patched.notpip._internal.resolution.resolvelib.factory:Cannot install flask~=2.0.1 and flask~=2.1.1 because these package versions have conflicting dependencies.
```

Pipfile:
```
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
flask = "~=2.0.1"
abcdef = {ref = "1.1.0", git = "https://***@github.com/abcdef-Utilities.git"}

[dev-packages]


[requires]
python_version = "3.7"
```
abcdef has flask ~=2.1.1 in setup.py

**Comment by matteius** [comment]
Linking to the issue report:  https://github.com/pypa/pipenv/issues/5428

I assume something broke the tilde operator -- will prioritize looking into this.
